src/middlewared/middlewared/plugins/iscsi_/alua.py

```diff
@@ -611,3 +613,49 @@ async def wait_for_alua_settled(self, sleep_interval=1, retries=10):
             await asyncio.sleep(sleep_interval)
             retries -= 1
         self.logger.warning('Gave up waiting for ALUA to settle')
+
+    @job(lock='force_close_sessions', transient=True, lock_queue_size=1)
+    async def force_close_sessions(self, job):
+        job.set_progress(0, 'Start force-close of iSCSI sessions')
+        self.logger.debug('Start force-close of iSCSI sessions')
+
+        await run('scst_util.sh', 'force-close')
+
+        job.set_progress(100, 'Complete force-close of iSCSI sessions')
+        self.logger.debug('Complete force-close of iSCSI sessions')
+
+    @job(lock='reset_active', transient=True, lock_queue_size=1)
+    async def reset_active(self, job):
+        """Job to be run on the ACTIVE node before the STANDBY node will join."""
+        job.set_progress(0, 'Start logout HA targets')
+        self.logger.debug('Start logout HA targets')
+
+        # This is similar, but not identical to iscsi.target.logout_ha_targets
+        # The main difference is these are logged out in series, to allow e.g. cluster_mode settle
+        # This is also why it is a job. it may take longer to run.
+        iqns = await self.middleware.call('iscsi.target.active_ha_iqns')
+
+        # Check what's already logged in
+        existing = await self.middleware.call('iscsi.target.logged_in_iqns')
+
+        # Generate the set of things we want to logout (don't assume every IQN, just the HA ones)
+        todo = set()
+        for iqn in iqns.values():
```

Comment:
This isn't needed but these 4 lines can be easily converted to a single line using a comprehension
```
todo = set(i for i in iqn.values() if i in existing)
```